### PR TITLE
JAMES-2261 Hotfix/imap utils mailbox nullcheck (community)

### PR DESCRIPTION
**Pull request by chibenwa**
Reopen of https://github.com/linagora/james-project/pull/1191 with a proper branch name

### CODE REVIEW

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
```

[13c5dbc2fe6ea5c6de8ccd1cb2eb94c5cf6172d9] [ERROR] Errors: 
[13c5dbc2fe6ea5c6de8ccd1cb2eb94c5cf6172d9] [ERROR]   Broken pipe (Write failed)
[13c5dbc2fe6ea5c6de8ccd1cb2eb94c5cf6172d9] [ERROR]   Broken pipe (Write failed)
[13c5dbc2fe6ea5c6de8ccd1cb2eb94c5cf6172d9] [INFO] 
[13c5dbc2fe6ea5c6de8ccd1cb2eb94c5cf6172d9] [ERROR] Tests run: 2586, Failures: 0, Errors: 2, Skipped: 2
```

test this please

File: protocols/imap/src/main/java/org/apache/james/imap/api/ImapSessionUtils.java [hunk 1]
```diff
@@ -33,13 +37,9 @@ public static MailboxSession getMailboxSession(ImapSession session) {
...
     public static String getUserName(ImapSession imapSession) {
-        final String result;
-        final MailboxSession mailboxSession = getMailboxSession(imapSession);
-        if (mailboxSession == null) {
-            result = null;
-        } else {
-            result = mailboxSession.getUser().getUserName();
-        }
-        return result;
+        Preconditions.checkNotNull(imapSession);
```
Comment (aduprat) [inline protocols/imap/src/main/java/org/apache/james/imap/api/ImapSessionUtils.java]: This _Precondition_ was not in the previous version.
It's not a problem for me, but you may introduce a test for that.

Comment (chibenwa) [reply protocols/imap/src/main/java/org/apache/james/imap/api/ImapSessionUtils.java]: It was a previous behavu=iour, I felt better if it's explicit. I'll do.

Comment (chibenwa) [reply protocols/imap/src/main/java/org/apache/james/imap/api/ImapSessionUtils.java]: The test is already here : **getUserNameShouldThrowOnNullImapSession**

**Pull request merged**